Misc/NEWS.d/next/IDLE/2021-05-16-16-24-31.bpo-40452.Nayfdf.rst

```diff
@@ -0,0 +1 @@
+fix IDLE clearing the clipboard upon exit
```

Comment:
With this PR, IDLE explicitly calls `_tkinter.finalize_tcl()`.

This PR doesn't fix tkinter; any other app using tkinter will still suffer from the clipboard being cleared upon exit. Until we (possibly) make this public in the future, IDLE should be the only app to benefit from this (and perhaps other apps using this undocumented, clearly non-public, function.)

Reply:
So, yes, the underlying problem is that tkinter currently doesn't properly finalize Tcl, leading to the clipboard being cleared. But this fix is IDLE specific, and as far as users are concerned, "IDLE not clearing the clipboard upon exit" is the change in behavior this PR results in.

Reply:
People (beginners) constantly blame IDLE for what they, or python, or ??? do.  I don't want to feed that.  Besides which, I checked and the clipboard is not absent unless one asks *tk* to save something there and do not copy anything from another app.  Open IDLE and close without copying and previous clipboard content remains.  Before tcl copies, the previous content is discarded, but the new content is not quite 'fixed' for post-exit external access.  Maybe the new content is not actually 'cleared'. So lets skip cause and blame and only mention specific problem and fix.
```suggestion
Fix new clipboard content copied from IDLE being gone after exiting IDLE.
```
This does not imply anything about exiting tkinter otherwise.  I agree, leave that for later after using the private fix a bit.